script.js

```diff
@@ -1,12 +1,363 @@
-//You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+const currentState = {
+  showsFetched: false,
+  episodesFetched: false,
+  allShows: null,
+  showSelected: null,
+  allEpisodes: null,
+  showsID: "",
+};
+
+const showDropDown = document.querySelector("#series__dropdown");
+const displayingFilterResults = createElement("p", "");
+let layoutSelector = document.querySelector(".layout");
+
+// Fetch API function to fetch data upon call
+async function fetchAPI(url) {
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw new Error(`Could not fetch data from ${url}`);
+  }
+  return response.json();
+}
+
+function addCardLayout() {
+  layoutSelector.classList.add("layout__for__cards");
+  searchField.classList.add("test");
+}
+
+function removeCardLayout() {
+  layoutSelector.classList.remove("layout__for__cards");
+}
+
+// Fetches shows data via API
+async function getAllShows() {
```

Comment:
What happens if the code fails to fetch the shows? Is the user informed about the failed action in any way?